### ISSUE TITLE
Subtitles in PIP

### ISSUE THREAD

**Issue by drikqlis** [jellyfin/jellyfin-web#2480]
**Describe the bug**
Subtitles in PIP are not displayed. Instead they are shown on inactive screen. 

**System (please complete the following information):**
 - OS: Ubuntu
 - Virtualization: Hyper-V
 - Clients: Chromium 88
 - Browser: Chromium 88
 - Jellyfin Version: 10.6.4
 - Playback: Direct Play
 - Reverse Proxy: nginx
 - Base URL: none

**To Reproduce**

1. Open wideo with subtitles.
2. Press the picture-in-picture button.
3. Subtitles are not displayed in PIP window.

**Expected behavior**
Subtitles should be displayed in PIP window.

**Comment by alex9099** [comment]
This is still relevant, seems like firefox (and likely other browsers) use WebVTT.
Some reference https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API

**Comment by thornbill** [comment]
I discussed this one recently on Matrix. It is an interesting problem... I believe we would need to switch out the video stream when entering either PiP or full screen playback (both encounter the same issue) to use either burned in subtitles or a format (webvtt) that is natively supported by the video element in browsers.

**Comment by alex9099** [comment]
Switching into burned in subs doesn't seem like the best option as it takes some time to load and gives a unnecessary load on the server, should only be used in last resort if the client doesn't support webvtt.

The best option I think would be extract the subtitles (in case they are embedded in an mkv for ex) and convert them to the PIP compatible format WebVTT

For reference, ffmpeg can convert subtitle formats:

`ffmpeg -i my_file.srt outfile.vtt`

From https://trac.ffmpeg.org/wiki/ExtractSubtitles

**Comment by davidngrc** [comment]
I check, you tube can show subtitle in picture in picture in firefox, but Jellyfin Web can't.
see the firefox discussion here 
https://bugzilla.mozilla.org/show_bug.cgi?id=1551058

**Comment by WatToDoWad** [comment]
> I discussed this one recently on Matrix. It is an interesting problem... I believe we would need to switch out the video stream when entering either PiP or full screen playback (both encounter the same issue) to use either burned in subtitles or a format (webvtt) that is natively supported by the video element in browsers.

Is there currently no way to maintain the same video stream going in and out of PiP, while having WebVTT subtitles (i.e. no burned in subs)?

**Comment by NewsGuyTor** [comment]
> > I discussed this one recently on Matrix. It is an interesting problem... I believe we would need to switch out the video stream when entering either PiP or full screen playback (both encounter the same issue) to use either burned in subtitles or a format (webvtt) that is natively supported by the video element in browsers.
> 
> Is there currently no way to maintain the same video stream going in and out of PiP, while having WebVTT subtitles (i.e. no burned in subs)?

No, it's pretty simple - you'd just need to use WebVTT for both non-PiP and PiP, like Youtube, instead of the custom system Jellyfin Web uses now.

There's really no good reason not to use WebVTT for the whole subtitle system on Jellyfin Web, except that the custom system is already developed and working well (except for PiP). To upgrade to WebVTT would get subtitles working in PiP, but obviously take some time and effort too.

I don't get why people are talking about burning in subtitles, that's a terrible solution (provided we are talking about text-based subtitles, not image based ones, ofc).

**Comment by casperghst42** [comment]
Using Jellyfin player on iPad.

I just extracet the embedded subtitles from an mkv - to srt. I then converted it to vtt. 

I then did a rescan of files, and jellyfin found the two new subtitle files, listed as SUBRIP and WEBVTT. I then selected the WEBVTT and started playing the file - everything good. I then selected PIP and no subtitles. Actually the subtitles were shown on the main window (which is otherwise empty).

I will continue to pay for my Emby as I do like to use PIP and I do like to watch foreign language content.

**Comment by thornbill** [comment]
@casperghst42 does it change anything if you set the Subtitle Styling option to "Native"?

![image](https://github.com/user-attachments/assets/f88d62c0-931e-42aa-ac23-8abee900a3b0)

**Comment by casperghst42** [comment]
> [@casperghst42](https://github.com/casperghst42) does it change anything if you set the Subtitle Styling option to "Native"?

No the same, subtitles are shown in the "main" window of the app. or browser.

<img width="798" height="467" alt="Image" src="https://github.com/user-attachments/assets/1a36afd4-7fa5-47ac-bd4b-6f7bb872bd15" />

**Comment by NewsGuyTor** [comment]
I've explained why it can't work with the current codebase right above

**Comment by casperghst42** [comment]
> I've explained why it can't work with the current codebase right above

You have and you also mention that VTT is the solution, I extracted the srt from the media and then converted it to a vtt - still same situation. 

I'll just use something else and come back in a while to see if anything have change.

**Comment by NewsGuyTor** [comment]
Ah, I meant the standard, not just the format. Jellyfin doesn't display VTT subs using a WebVTT compliant renderer.

**Comment by casperghst42** [comment]
> Ah, I meant the standard, not just the format. Jellyfin doesn't display VTT subs using a WebVTT compliant renderer.

No worries, I changed it. Nothing changed, no subs (no matter the type) is shown in PiP. I'll just wait until it's changed, and use something else for my foreign (non-english) media.

**Comment by mrtimothyduong** [comment]
I've tested with fladder (https://github.com/DonutWare/Fladder), you can use the web version from this repo to connect to your local jellyfin instance. Of course probably use a burner account.

Example: Left = Fladder, Right = Jellyfin builtin Web player

<img width="1558" height="701" alt="Image" src="https://github.com/user-attachments/assets/6ff6cc73-8613-4b82-8690-b3e78679eb68" />

**Comment by EVOTk** [comment]
Hello
Same issue for me !

**Comment by ShivaSingularity** [comment]
any workaround?

**Comment by mrtimothyduong** [comment]
@ShivaSingularity read my post.

- Workaround for Jellyfin-web? No
- Workaround for 3rd Party Web App? Yes, something like Fladder
- Workaround for 3rd Party App? Yes, Infuse.
